File: Лабы_1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void print_row(int spaces, int bricks);

int main(void)
{
    int height;
    do
    {
        height = get_int("Введите высоту пирамиды: ");
    }
    while (8 < height < 1);

    for (int i = 0; i < height; i++)
    {
        print_row(height - i, i + 1);
    }
}

void print_row(int spaces, int bricks)
{
    for (int i = 0; i < spaces - 1; i++)
    {
        printf(" ");
    }
    for (int j = 0; j < bricks; j++)
    {
        printf("#");
    }
    printf("  ");
    for (int j = 0; j < bricks; j++)
    {
        printf("#");
    }
    printf("\n");
}

File: Лабы_6/sentimental-mario-less/mario.py
from cs50 import get_int

def main():
    while True:
        height = get_int("Height: ")
        if height > 0 and height < 9:
            break

    # Вывод строк с кирпичами
    for i in range(height):
        print_row(height - i, i + 1)

def print_row(spaces, bricks):
    # Вывод пробелов
    for i in range(spaces - 1):
        print(" ", end="")

    # Вывод кирпичей
    for j in range(bricks):
        print("#", end="")

    print()

main()
